### PR TITLE
Derive: second virial coefficient of van der Waals gas

### PR DESCRIPTION
**Pull request by alesanter**

### CODE REVIEW

File: symplyphysics/laws/thermodynamics/equations_of_state/van_der_waals/second_virial_coefficient.py [hunk 1]
```diff
@@ -39,7 +43,48 @@
...
+
+_compressibility_via_volume = compressibility_def.definition.rhs.subs({
+    compressibility_def.pressure: _pressure_expr,
+    compressibility_def.temperature: temperature,
+})
+
+_compressibility_via_density = solve(
+    [
+        Eq(_compressibility_sym, _compressibility_via_volume),
+        Eq(_molar_density, _mole_count / _volume),
```
Comment (blackyblack) [inline symplyphysics/laws/thermodynamics/equations_of_state/van_der_waals/second_virial_coefficient.py:69]: Might be a good idea to also generalize it as some `volumetric_density` of something law.

File: symplyphysics/laws/thermodynamics/equations_of_state/van_der_waals/second_virial_coefficient.py [hunk 2]
```diff
@@ -39,7 +43,48 @@
...
+    ],
+    (
+        _compressibility_sym,
+        _mole_count,
+    ),
+    dict=True,
+)[0][_compressibility_sym]
+
+_compressibility_series = _compressibility_via_density.series(_molar_density, 0, 2).removeO()
```
Comment (blackyblack) [inline symplyphysics/laws/thermodynamics/equations_of_state/van_der_waals/second_virial_coefficient.py:78]: This assumption should be added to conditions.

File: symplyphysics/laws/thermodynamics/equations_of_state/van_der_waals/second_virial_coefficient.py [hunk 3]
```diff
@@ -39,7 +43,48 @@
...
+
+_virial_eqn = virial_eqn.law.rhs.subs(virial_eqn.third_virial_coefficient, 0)
```
Comment (blackyblack) [inline symplyphysics/laws/thermodynamics/equations_of_state/van_der_waals/second_virial_coefficient.py:80]: Why?

Comment (alesanter) [reply symplyphysics/laws/thermodynamics/equations_of_state/van_der_waals/second_virial_coefficient.py:80]: We're interested in the virial expansion up to the second order of density inclusive, so there's no need to include the third coefficient. I'll write about it

File: symplyphysics/laws/thermodynamics/equations_of_state/van_der_waals/second_virial_coefficient.py (hunk 2 above)
Comment (alesanter) [reply symplyphysics/laws/thermodynamics/equations_of_state/van_der_waals/second_virial_coefficient.py:78]: Agreed

File: symplyphysics/laws/thermodynamics/equations_of_state/van_der_waals/second_virial_coefficient.py [hunk 4]
```diff
@@ -39,7 +46,36 @@
...
+
+_mole_count_expr = density_qty_law.law.rhs.subs({
+    density_qty_law.volumetric_density: _molar_density,
+    density_qty_law.volume: _volume,
+})
+
+_compressibility_via_density = _compressibility_via_volume.subs(_mole_count, _mole_count_expr).simplify()
+
+# Using the formula for Taylor series coefficients
+_second_virial_coefficient = _compressibility_via_density.diff(_molar_density, 1).subs(_molar_density, 0)
```
Comment (blackyblack) [inline symplyphysics/laws/thermodynamics/equations_of_state/van_der_waals/second_virial_coefficient.py:76]:
```suggestion
_second_virial_coefficient = _compressibility_via_density.diff(_molar_density).subs(_molar_density, 0)
```

File: symplyphysics/laws/thermodynamics/equations_of_state/van_der_waals/second_virial_coefficient.py [hunk 5]
```diff
@@ -39,7 +46,36 @@
...
+})
+
+_mole_count_expr = density_qty_law.law.rhs.subs({
+    density_qty_law.volumetric_density: _molar_density,
+    density_qty_law.volume: _volume,
+})
+
+_compressibility_via_density = _compressibility_via_volume.subs(_mole_count, _mole_count_expr).simplify()
+
+# Using the formula for Taylor series coefficients
```
Comment (blackyblack) [inline symplyphysics/laws/thermodynamics/equations_of_state/van_der_waals/second_virial_coefficient.py:75]: Elaborate, how second virial coefficient is related to Taylor series and derivative of compressibility.

**Pull request merged**